glean-core/android/src/main/java/mozilla/telemetry/glean/Dispatchers.kt

```diff
@@ -111,13 +116,28 @@ internal object Dispatchers {
             // NOTE: This has the potential for causing a task to execute out of order in certain
             // situations. If a library or thread that runs before init happens to record
             // between when the queueInitialTasks is set to false and the taskQueue finishing
-            // launching, then that task could be executed out of the queued order. See the bugzilla
-            // bug tracking this for more info: https://bugzilla.mozilla.org/show_bug.cgi?id=1568503
-            queueInitialTasks = false
-            taskQueue.forEach { task ->
-                task.invoke()
+            // launching, then that task could be executed out of the queued order.
+            val job = coroutineScope.launch {
+                taskQueue.forEach { task ->
+                    task.invoke()
+                }
+                queueInitialTasks = false
+                taskQueue.clear()
+            }
+
+            // In order to ensure that the queued tasks are executed in the proper order, we will
+            // wait up to 5 seconds for it to complete, otherwise we will reset the flag so that
+            // new tasks may continue to run.
+            runBlocking {
+                val result = withTimeoutOrNull(QUEUE_PROCESSING_TIMEOUT_MS) {
```

Comment:
What is the return value of this block when it doesn't time out?

Reply:
It returns the coroutine back.  See [here](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/with-timeout-or-null.html)